### PR TITLE
Add ability for user defined profession names

### PR DESCRIPTION
**Pull request by CodeBandit**
#### Summary
SUMMARY: Content "Add ability for user-defined profession names"

#### Purpose of change
It has no impact on gameplay other than immersion. After all, we don't want our characters to forever be defined with "tweaker", "naked and afraid", or "lost submissive".

#### Describe the solution

Instead of this:
image

we can press * during character display to get a popup below
image

We can define our characters as this instead:
image

Deleting the custom profession will reset it back to default functionality.

#### Describe alternatives you've considered
None.

#### Testing
Add and remove the custom profession.
Save and reload for save game persistence.

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -269,6 +269,8 @@ player::player()
     prof = profession::has_initialized() ? profession::generic() :
            nullptr; //workaround for a potential structural limitation, see player::create
 
+    custom_profession = "";
```
Comment (kevingranade) [inline src/player.cpp:272]: std::string is initialized empty, this is redundant.

File: src/player_display.cpp [hunk 2]
```diff
@@ -1326,27 +1327,33 @@ void player::disp_info()
...
-        std::string race;
-        for( const trait_id &mut : get_mutations() ) {
-            const mutation_branch &mdata = mut.obj();
-            if( mdata.threshold ) {
-                race = mdata.name();
-                break;
+    // Post-humanity trumps your pre-Cataclysm life
+    // Unless you have a custom profession.
+    std::string race;
+    if( g->u.custom_profession.empty() || g->u.custom_profession == "" ) {
```
Comment (kevingranade) [inline src/player_display.cpp:1333]: These are the same check.
```suggestion
    if( g->u.custom_profession.empty() ) {
```

File: src/player_display.cpp [hunk 3]
```diff
@@ -1402,6 +1410,57 @@ void player::disp_info()
...
+            werase( w_tip );
+
+            g->refresh_all();
+
+            draw_grid_borders( w_grid_top, w_grid_skill, w_grid_trait, w_grid_bionics, w_grid_effect,
+                               info_win_size_y, infooffsetybottom, skill_win_size_y + 1, trait_win_size_y + 1,
+                               bionics_win_size_y + 1, effect_win_size_y + 1 );
+
+            // Print name and header
+            if( g->u.custom_profession.empty() || g->u.custom_profession == "" ) {
```
Comment (kevingranade) [inline src/player_display.cpp:1434]: Redundant again
```suggestion
            if( g->u.custom_profession.empty() ) {
```

**Comment by CodeBandit** [comment]
I have removed the redundant inits and checks

**Pull request merged**